m4opt/models/math.py

```diff
@@ -0,0 +1,247 @@
+              **kwargs) -> Union[Quantity, float]:
+    """Integrate a 1D model using adaptive trapezoidal quadrature.
+
+    Parameters
+    ----------
+    model : :class:`astropy.modeling.Model`
+        The model to integrate.
+    quick_and_dirty_npts : int, optional
+        Number of sample points. If provided, disable adaptive quadrature and
+        instead use fixed-order quadrature with the specified number of
+        regularly spaced sample points.
+    **kwargs : dict
+        Additional keyword arguments passed to
+        :meth:`scipy.integrate.quad_vec`.
+
+    Returns
+    -------
+    integral : float, :class:`astropy.units.Quantity`
+        The integral over the bounding box of the model's inputs.
+
+    Notes
+    -----
+    When integrating compound models, the domain of integration is subdivided
+    using the bounding box endpoints of all of the submodels. The integrator
+    minimizes the error by tracking the error contributions from all of the
+    subdivisions. The integrator can be slow; passing lower tolerances can
+    speed up the calculation with a loss of accuracy. For a faster version
+    without convergence checking, try the `quick_and_dirty_npts` option.
+
+    Examples
+    --------
+
+    You can integrate dimensionless models:
+
+    >>> from astropy.modeling import models
+    >>> from astropy import units as u
+    >>> from m4opt.models import integrate
+    >>> import numpy as np
+    >>> model = models.Box1D(width=3)
+    >>> integrate(model)
+    3.0
+    >>> integrate(model, quick_and_dirty_npts=10000)
+    3.0
+    >>> model = models.Gaussian1D() * models.Const1D(1 / np.sqrt(2 * np.pi))
+    >>> integrate(model, epsrel=1e-7)
+    0.9999999648585338
+    >>> integrate(model, quick_and_dirty_npts=10000)
+    0.9999999620207557
+
+    Or models with dimensions:
+
+    >>> model = models.Lorentz1D(x_0=1 * u.micron, fwhm=0.1 * u.micron,
+    ...                          amplitude=1 * u.erg / u.micron)
+    >>> integrate(model, epsrel=1e-7)
+    <Quantity 0.1550799 erg>
+    >>> integrate(model, quick_and_dirty_npts=10000)
+    <Quantity 0.1550799 erg>
+
+    """
+    x_unit = _get_1d_units_for_dict(model.input_units)
+
+    intervals = _get_intervals(model)
+    a, *points, b = np.unique(Quantity(list(chain.from_iterable(
+        (i.lower, i.upper) for i in intervals)), x_unit).value.ravel())
+
+    # FIXME: Cannot rely on model.return_units because it is not set for all
+    # models (e.g. Lorentz1D).
+    y_unit = getattr(model(a * x_unit), 'unit', None) or 1
+
+    def func(x):
+        result = model(x * x_unit)
+        if hasattr(result, 'to_value'):
+            result = result.to_value()
+        return result
+
+    if quick_and_dirty_npts is not None:
+        x = np.linspace(a, b, quick_and_dirty_npts)
+        yint = np.trapz(func(x), x)
+    else:
+        yint, _ = quad_vec(func, a, b, points=points,
+                           quadrature='trapezoid', **kwargs)
```

Comment:
maybe
```suggestion
        kwargs.setdefault('quadrature', 'trapezoid')
        yint, _ = quad_vec(func, a, b, points=points, **kwargs)
```
just so that the user has the option to re-set this if they really want to?